core/jobs/src/main/java/com/cognifide/aet/job/common/comparators/source/SourceComparator.java

```diff
@@ -119,4 +120,16 @@ private void addFormattedSources(ComparatorStepResult result) {
     result.addData("formattedSource", data);
     result.addData("sourceCompareType", sourceCompareType.name());
   }
+
+  private void addTimestampToResult(ComparatorStepResult result) {
+    long stamp;
+    try {
+      stamp = artifactsDAO.getArtifactUploadDate(properties, properties.getPatternId()).getTime();
+    } catch(Exception e) {
+      stamp = 0;
+    }
+    result.addData("patternTimestamp", Long.toString(stamp));
```

Comment:
This should be done a bit differently: instead of doing it inside try-catch and zeroing `stamp`, you should put the `result.addData("patternTimestamp", ...)` inside `try` block (with properly labelled noop/debug log for `catch`), and then conditionally display it inside the view of report.

Reply:
done